src/main/scala/com/advancedtelematic/campaigner/http/CampaignResource.scala

```diff
@@ -79,14 +96,19 @@ class CampaignResource(extractAuth: Directive1[AuthedNamespaceScope], director:
               complete(campaigns.update(campaign.id, updated.name, updated.metadata.toList.flatten.map(_.toCampaignMetadata(campaign.id))))
             }
           } ~
-          (post & path("launch")) {
-            complete(campaigns.launch(campaign.id))
+          post {
+            path("launch") {
+              complete(campaigns.launch(campaign.id))
+            } ~
+            path("cancel") {
+              complete(campaigns.cancel(campaign.id))
+            } ~
+            (path("retry-failed") & entity(as[RetryFailedDevices])) { request =>
```

Comment:
Maybe this is not the best name for the endpoint because we're not retrying yet, just creating a retry-campaign. Two ideas: rename the endpoint to reflect that, or create the retry-campaign and launch it already. In any case we still return the campaignId to the FE. Any preference?

Reply:
I'd vote for launching at the time of creation because it's gonna be done by FE anyway with a next request.